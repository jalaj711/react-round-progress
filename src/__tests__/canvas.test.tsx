import React from "react";
import ReactDOM from 'react-dom/client';
import { act } from 'react-dom/test-utils';
import { describe, expect, test } from '@jest/globals';
import 'jest-canvas-mock';

import CircularProgress from '../RoundProgress'

const CANVASID: string = "id";
let root: any = null;
let canvas: any = null;
let ctx: any = null;
let container: any = null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
  root = ReactDOM.createRoot(
    container as HTMLElement
  );
});

afterEach(() => {
  canvas = document.getElementById(CANVASID);
  if (canvas) {
    ctx = canvas.getContext("2d");

    const events = ctx.__getEvents();
    expect(events).toMatchSnapshot();
  }

  // cleanup on exiting
  act(() => {
    root.unmount();
  });
});

describe('CircularProgress', () => {
  test('should call these canvas functions', async () => {
    act(() => {
      root.render(
        <CircularProgress
          value={60}
          id={CANVASID}
          range={240}
          centralNeedle={true}
          gaugeMarks={true}
          gaugeMarkSeperation={240 / 10}
          showValue={true}
          dimension={250}
          lineWidth={4}
          offSet={150}
          colors={{
            stroke: "blue",
            gaugeMarks: "red",
            needle: "green",
            backgroundTrack: "grey",
          }}
        />
      );
    });

    canvas = document.getElementById("id");
    ctx = canvas.getContext("2d");

    expect(ctx.clearRect).toBeCalled();
    expect(ctx.beginPath).toBeCalled();
    expect(ctx.arc).toBeCalled();
    expect(ctx.stroke).toBeCalled();
    expect(ctx.moveTo).toBeCalled();
    expect(ctx.lineTo).toBeCalled();

    expect(ctx.scale).not.toBeCalled();
  });

  test('should not call some canvas functions for no gaugeMarks, centralNeedle', async () => {
    act(() => {
      root.render(
        <CircularProgress
          value={60}
          id={CANVASID}
          range={240}
          centralNeedle={false}
          gaugeMarks={false}
          gaugeMarkSeperation={240 / 10}
          showValue={true}
          dimension={250}
          lineWidth={4}
          offSet={150}
          colors={{
            stroke: "blue",
            gaugeMarks: "red",
            needle: "green",
            backgroundTrack: "grey",
          }}
        />
      );
    });

    canvas = document.getElementById("id");
    ctx = canvas.getContext("2d");

    expect(ctx.clearRect).toBeCalled();
    expect(ctx.beginPath).toBeCalled();
    expect(ctx.arc).toBeCalled();
    expect(ctx.stroke).toBeCalled();

    expect(ctx.moveTo).not.toBeCalled();
    expect(ctx.lineTo).not.toBeCalled();
    expect(ctx.scale).not.toBeCalled();
  });
});


describe('Errors when invalid props', () => {
  test('should throw error', async () => {
    expect(() => {
      act(() => {
        root.render(
          <CircularProgress
            value={60}
            id={CANVASID}
            dimension={2}
            lineWidth={4}
          />
        );
      });

      canvas = document.getElementById("id");
      ctx = canvas.getContext("2d");
    }).toThrow(DOMException);
  });
});

