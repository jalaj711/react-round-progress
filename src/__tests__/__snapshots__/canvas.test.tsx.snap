// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgress should call these canvas functions 1`] = `
[
  {
    "props": {
      "height": 250,
      "width": 250,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "value": "round",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineCap",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "anticlockwise": false,
      "endAngle": 6.806784082777885,
      "radius": 121,
      "startAngle": 2.6179938779914944,
      "x": 125,
      "y": 125,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "arc",
  },
  {
    "props": {
      "value": "#808080",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 4,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "anticlockwise": false,
            "endAngle": 6.806784082777885,
            "radius": 121,
            "startAngle": 2.6179938779914944,
            "x": 125,
            "y": 125,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "arc",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "x": 28.005154776142874,
      "y": 181,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  {
    "props": {
      "x": 23.67502775722069,
      "y": 183.5,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  {
    "props": {
      "value": "#ff0000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "x": 28.005154776142874,
            "y": 181,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        {
          "props": {
            "x": 23.67502775722069,
            "y": 183.5,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "x": 13.613547718753395,
      "y": 136.7071878859772,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  {
    "props": {
      "x": 8.640938241912025,
      "y": 137.22983020231544,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  {
    "props": {
      "value": "#ff0000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "x": 13.613547718753395,
            "y": 136.7071878859772,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        {
          "props": {
            "x": 8.640938241912025,
            "y": 137.22983020231544,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "x": 18.481670174942792,
      "y": 90.3900966300059,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  {
    "props": {
      "x": 13.726387593467024,
      "y": 88.84501165813117,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  {
    "props": {
      "value": "#ff0000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "x": 18.481670174942792,
            "y": 90.3900966300059,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        {
          "props": {
            "x": 13.726387593467024,
            "y": 88.84501165813117,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "x": 41.76777954653184,
      "y": 50.05737208780788,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  {
    "props": {
      "x": 38.05205541914488,
      "y": 46.71171905601359,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  {
    "props": {
      "value": "#ff0000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "x": 41.76777954653184,
            "y": 50.05737208780788,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        {
          "props": {
            "x": 38.05205541914488,
            "y": 46.71171905601359,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "x": 79.44549597551035,
      "y": 22.68290874402871,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  {
    "props": {
      "x": 77.41181276013134,
      "y": 18.115181455815716,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  {
    "props": {
      "value": "#ff0000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "x": 79.44549597551035,
            "y": 22.68290874402871,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        {
          "props": {
            "x": 77.41181276013134,
            "y": 18.115181455815716,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "x": 124.99999999999999,
      "y": 13,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  {
    "props": {
      "x": 124.99999999999999,
      "y": 8,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  {
    "props": {
      "value": "#ff0000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "x": 124.99999999999999,
            "y": 13,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        {
          "props": {
            "x": 124.99999999999999,
            "y": 8,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "x": 170.55450402448963,
      "y": 22.68290874402871,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  {
    "props": {
      "x": 172.58818723986863,
      "y": 18.1151814558157,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  {
    "props": {
      "value": "#ff0000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "x": 170.55450402448963,
            "y": 22.68290874402871,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        {
          "props": {
            "x": 172.58818723986863,
            "y": 18.1151814558157,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "x": 208.2322204534682,
      "y": 50.05737208780789,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  {
    "props": {
      "x": 211.94794458085514,
      "y": 46.71171905601359,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  {
    "props": {
      "value": "#ff0000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "x": 208.2322204534682,
            "y": 50.05737208780789,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        {
          "props": {
            "x": 211.94794458085514,
            "y": 46.71171905601359,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "x": 231.5183298250572,
      "y": 90.39009663000587,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  {
    "props": {
      "x": 236.27361240653295,
      "y": 88.84501165813114,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  {
    "props": {
      "value": "#ff0000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "x": 231.5183298250572,
            "y": 90.39009663000587,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        {
          "props": {
            "x": 236.27361240653295,
            "y": 88.84501165813114,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "x": 236.38645228124662,
      "y": 136.70718788597713,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  {
    "props": {
      "x": 241.359061758088,
      "y": 137.2298302023154,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  {
    "props": {
      "value": "#ff0000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "x": 236.38645228124662,
            "y": 136.70718788597713,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        {
          "props": {
            "x": 241.359061758088,
            "y": 137.2298302023154,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "x": 221.9948452238571,
      "y": 181,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  {
    "props": {
      "x": 226.32497224277932,
      "y": 183.5,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  {
    "props": {
      "value": "#ff0000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "x": 221.9948452238571,
            "y": 181,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        {
          "props": {
            "x": 226.32497224277932,
            "y": 183.5,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "x": 125,
      "y": 125,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  {
    "props": {
      "x": 168.52082080911063,
      "y": 27.2506360322417,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  {
    "props": {
      "value": "#008000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 4,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "x": 125,
            "y": 125,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        {
          "props": {
            "x": 168.52082080911063,
            "y": 27.2506360322417,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "value": "#0000ff",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 5.2,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "anticlockwise": false,
      "endAngle": 5.1312680008633285,
      "radius": 121,
      "startAngle": 2.6179938779914944,
      "x": 125,
      "y": 125,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "arc",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "anticlockwise": false,
            "endAngle": 5.1312680008633285,
            "radius": 121,
            "startAngle": 2.6179938779914944,
            "x": 125,
            "y": 125,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "arc",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
]
`;

exports[`CircularProgress should not call some canvas functions for no gaugeMarks, centralNeedle 1`] = `
[
  {
    "props": {
      "height": 250,
      "width": 250,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "value": "round",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineCap",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "anticlockwise": false,
      "endAngle": 6.806784082777885,
      "radius": 121,
      "startAngle": 2.6179938779914944,
      "x": 125,
      "y": 125,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "arc",
  },
  {
    "props": {
      "value": "#808080",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 4,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "anticlockwise": false,
            "endAngle": 6.806784082777885,
            "radius": 121,
            "startAngle": 2.6179938779914944,
            "x": 125,
            "y": 125,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "arc",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "value": "#0000ff",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 5.2,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "anticlockwise": false,
      "endAngle": 5.1312680008633285,
      "radius": 121,
      "startAngle": 2.6179938779914944,
      "x": 125,
      "y": 125,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "arc",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "anticlockwise": false,
            "endAngle": 5.1312680008633285,
            "radius": 121,
            "startAngle": 2.6179938779914944,
            "x": 125,
            "y": 125,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "arc",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
]
`;
